### ISSUE TITLE
Add a dictionary to a collection

### ISSUE THREAD

**Issue by FrancescoSaverioZuppichini** [tensorflow/tensorflow#20475]
### System information
OS Platform and Distribution Mac OS X latest
TensorFlow installed from pip
TensorFlow version 1.8
Bazel version N/A
CUDA/cuDNN version N/A
GPU model and memory N/A
Exact command to reproduce see code below

### Describe the problem
Dear All,

I am looking for a way to store and retrieve a dictionary of tensors. If I try to pass a dict to `ft.add_to_collection` I get a warning and when I load the graph again the collection does not exist. This is a simple example:

```
import tensorflow as tf
tf.reset_default_graph()

foo  = tf.Variable([1,2,3], dtype=tf.float32)
boo  = tf.Variable([1,2,3], dtype=tf.float32)

tf.add_to_collection('test', {'foo' : foo, 'boo': [boo] })
tf.add_to_collection('test2', foo)

with tf.Session() as sess:
    saver = tf.train.Saver()
    sess.run(tf.global_variables_initializer())
    saver.save(sess,'save/model.ckpt')

tf.reset_default_graph() # reset so we are sure to load a new graph

with tf.Session() as sess:
    saver = tf.train.import_meta_graph("save/model.ckpt.meta")
    saver.restore(sess,'save/model.ckpt')
    graph = tf.get_default_graph()
    print(graph.get_collection('test'))
    print(graph.get_collection('test2'))
```

Output:

```
WARNING:tensorflow:Error encountered when serializing test.
Type is unsupported, or the types of the items don't match field type in CollectionDef.
'dict' object has no attribute 'name'
INFO:tensorflow:Restoring parameters from save/model.ckpt
[]
[<tf.Tensor 'Variable:0' shape=(3,) dtype=float32_ref>]
```

As you can see, after loading the graph, the `test` collection is empty.

Is there any way to properly store/retrieve a dictionary that maps tensors?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by FrancescoSaverioZuppichini** [comment]
Done it!

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 17 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 32 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by FrancescoSaverioZuppichini** [comment]
Any news?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by FrancescoSaverioZuppichini** [comment]
rip

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 53 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 68 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 85 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by Saduf2019** [comment]
@FrancescoSaverioZuppichini 
Could you please try on latest stable version of tf and let us know if this is still an issue.Thanks!

**Comment by FrancescoSaverioZuppichini** [comment]
I switched to PyTorch years ago, you should do the same :)

**Comment by Saduf2019** [comment]
@FrancescoSaverioZuppichini 
Please move this to closed status, if it is not a issue for you anymore.
